homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,644 @@
+            "Item"
+        ]
+        if isinstance(self.dict_of_deflection, list):
+            self.dict_of_deflection = self.dict_of_deflection[self.id]
+
+        _LOGGER.debug(
+            "Specific %s response: NewDeflectionList=%s",
+            SWITCH_TYPE_DEFLECTION,
+            self.dict_of_deflection,
+        )
+
+        self._attr_is_on = self.dict_of_deflection["Enable"] == "1"
+        self._is_available = True
+
+        self._attributes["Type"] = self.dict_of_deflection["Type"]
+        self._attributes["Number"] = self.dict_of_deflection["Number"]
+        self._attributes["DeflectionToNumber"] = self.dict_of_deflection[
+            "DeflectionToNumber"
+        ]
+        # Return mode sample: "eImmediately"
+        self._attributes["Mode"] = self.dict_of_deflection["Mode"][1:]
+        self._attributes["Outgoing"] = self.dict_of_deflection["Outgoing"]
+        self._attributes["PhonebookID"] = self.dict_of_deflection["PhonebookID"]
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle deflection switch."""
+        await async_service_call_action(
+            self._fritzbox_tools,
+            "X_AVM-DE_OnTel",
+            "1",
+            "SetDeflectionEnable",
+            NewDeflectionId=self.id,
+            NewEnable="1" if turn_on else "0",
+        )
+
+
+class FritzBoxProfileSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools DeviceProfile switch."""
+
+    def __init__(
+        self, fritzbox_tools: FritzBoxTools, device_friendly_name: str, profile: str
+    ) -> None:
+        """Init Fritz profile."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+        self.profile = profile
+
+        switch_info = SwitchInfo(
+            description=f"Profile {profile}",
+            friendly_name=device_friendly_name,
+            icon="mdi:router-wireless-settings",
+            type=SWITCH_TYPE_DEVICEPROFILE,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self._fritzbox_tools, device_friendly_name, switch_info)
+
+    async def _async_fetch_update(self) -> None:
+        """Update data."""
+        try:
+            status = await self.hass.async_add_executor_job(
+                self._fritzbox_tools.fritz_profiles[self.profile].get_state
+            )
+            _LOGGER.debug(
+                "Specific %s response: get_State()=%s",
+                SWITCH_TYPE_DEVICEPROFILE,
+                status,
+            )
+            if status == "never":
+                self._attr_is_on = False
+                self._is_available = True
+            elif status == "unlimited":
+                self._attr_is_on = True
+                self._is_available = True
+            else:
+                self._is_available = False
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Can we get rid of the broad except handler ?

Reply:
Unfortunately not, `fritzprofiles` library currently has a very basic exception handling